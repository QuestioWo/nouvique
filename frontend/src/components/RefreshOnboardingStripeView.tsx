import React from 'react';

import { Result } from 'neverthrow';
import { RouteComponentProps } from 'react-router';

import BasePage from './elements/BasePage';

import { UserStripeOnboardingLinkREST, resolveGETCall, PageProps, userStripeOnboardingLinkRESTLink } from '../utils';

import './RefreshOnboardingStripeView.css';

interface MatchParams {
	username: string
}

export interface RefreshOnboardingStripeViewProps extends RouteComponentProps<MatchParams>, PageProps { }

export class RefreshOnboardingStripeView extends React.Component<RefreshOnboardingStripeViewProps> {
	async componentDidMount() {
		const pathStripeOnboardingLink: string = userStripeOnboardingLinkRESTLink + this.props.match.params.username + '/';
		const resultStripeOnboardingLink: Result<UserStripeOnboardingLinkREST, Error> = await resolveGETCall<UserStripeOnboardingLinkREST>(pathStripeOnboardingLink, true);

		resultStripeOnboardingLink
			.map(res => {
				window.location.href = res.onboarding_link;

				return null; // necessary to silence warning
			})
			.mapErr(err => {
				// cannot be authenticated with the user therefore, jwt tokens must not be correct, therefore, deny form access
				console.error(err);

				this.props.updateAlertBar("User is not valid to access their Stripe account. Please re-log-in and try again", "danger", true);

				this.props.history.push('/');
			});
	}

	render() {
		return (
			<BasePage {...this.props}>
				<React.Fragment>
					Attempting to refresh Stripe onboarding link...
				</React.Fragment>
			</BasePage>
		)
	}
}