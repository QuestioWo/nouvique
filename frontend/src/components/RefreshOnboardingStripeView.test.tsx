import * as React from 'react';

import { mount, shallow } from 'enzyme';

import { RefreshOnboardingStripeView, RefreshOnboardingStripeViewProps } from './RefreshOnboardingStripeView';
import { MemoryRouter as Router } from 'react-router';

describe('RefreshOnboardingStripeView', () => {
	const getProps = (): RefreshOnboardingStripeViewProps => (
		{
			history: {} as any,
			location: {} as any,
			match: {} as any,
			updateAlertBar: {} as any,
			emptyBasket: {} as any,
			addToBasket: {} as any,
			removeFromBasket: {} as any,
			getBasketItems: {} as any,
			getTotalBasketPrice: {} as any,
			checkBasketAvailabilities: {} as any,
			stripePromise: {} as any,
			alert: {} as any
		}
	)

	const getComponent = (props: RefreshOnboardingStripeViewProps) => (
		<RefreshOnboardingStripeView {...props} />
	);


	it('Should render correctly with shallow', () => {
		const component = shallow(
			getComponent(getProps())
		);
		expect(component.debug()).toMatchSnapshot();
	});

	it('Should render correctly with mount', () => {
		// have to include router as will fail with react-router problem due to use of specific functions
		const component = mount(
			<Router>
				{getComponent(getProps())}
			</Router>
		);
		expect(component.debug()).toMatchSnapshot();
	});
});