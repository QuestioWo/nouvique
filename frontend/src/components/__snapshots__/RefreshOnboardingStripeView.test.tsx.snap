// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefreshOnboardingStripeView Should render correctly with mount 1`] = `
"<MemoryRouter>
  <Router history={{...}}>
    <RefreshOnboardingStripeView history={{...}} location={{...}} match={{...}} updateAlertBar={[Function: updateAlertBar]} emptyBasket={[Function: emptyBasket]} addToBasket={[Function: addToBasket]} removeFromBasket={[Function: removeFromBasket]} getBasketItems={[Function: getBasketItems]} getTotalBasketPrice={[Function: getTotalBasketPrice]} checkBasketAvailabilities={[Function: checkBasketAvailabilities]} stripePromise={{...}} alert={{...}}>
      <BasePage history={{...}} location={{...}} match={{...}} updateAlertBar={[Function: updateAlertBar]} emptyBasket={[Function: emptyBasket]} addToBasket={[Function: addToBasket]} removeFromBasket={[Function: removeFromBasket]} getBasketItems={[Function: getBasketItems]} getTotalBasketPrice={[Function: getTotalBasketPrice]} checkBasketAvailabilities={[Function: checkBasketAvailabilities]} stripePromise={{...}} alert={{...}}>
        <NavBar>
          <Navbar bg=\\"transparent\\" variant=\\"light\\" expand={true} collapseOnSelect={false}>
            <nav className=\\"navbar navbar-expand navbar-light bg-transparent\\">
              <withRouter(LinkContainer) to=\\"/\\">
                <LinkContainer to=\\"/\\" history={{...}} location={{...}} match={{...}} staticContext={[undefined]} replace={false} exact={false} strict={false} activeClassName=\\"active\\">
                  <Route path=\\"/\\" exact={false} strict={false}>
                    <NavbarBrand className=\\"active\\" style={{...}} href=\\"/\\" onClick={[Function (anonymous)]}>
                      <a style={{...}} href=\\"/\\" onClick={[Function (anonymous)]} className=\\"active navbar-brand\\">
                        <img alt=\\"\\" src=\\"/logo192.png\\" className=\\"nav-bar-logo d-inline-block align-top\\" />
                         
                        Nouvique
                      </a>
                    </NavbarBrand>
                  </Route>
                </LinkContainer>
              </withRouter(LinkContainer)>
              <NavbarCollapse className=\\"justify-content-end\\">
                <ForwardRef in={false} className=\\"justify-content-end\\" timeout={300} mountOnEnter={false} unmountOnExit={false} appear={false} getDimensionValue={[Function: getDefaultDimensionValue]}>
                  <ForwardRef addEndListener={[Function: transitionEndListener]} in={false} timeout={300} mountOnEnter={false} unmountOnExit={false} appear={false} aria-expanded={{...}} onEnter={[Function (anonymous)]} onEntering={[Function (anonymous)]} onEntered={[Function (anonymous)]} onExit={[Function (anonymous)]} onExiting={[Function (anonymous)]} childRef={{...}}>
                    <Transition in={false} timeout={300} mountOnEnter={false} unmountOnExit={false} appear={false} aria-expanded={{...}} onEnter={[Function (anonymous)]} onEntered={[Function (anonymous)]} onEntering={[Function (anonymous)]} onExit={[Function (anonymous)]} onExited={[Function (anonymous)]} onExiting={[Function (anonymous)]} addEndListener={[Function (anonymous)]} nodeRef={{...}} enter={true} exit={true}>
                      <div className=\\"justify-content-end navbar-collapse collapse\\" aria-expanded={{...}}>
                        <Nav className=\\"mr-auto\\" justify={false} fill={false}>
                          <ForwardRef as=\\"div\\" activeKey={[undefined]} className=\\"mr-auto navbar-nav\\" onSelect={[Function (anonymous)]}>
                            <div className=\\"mr-auto navbar-nav\\" onKeyDown={[Function: handleKeyDown]} role={[undefined]}>
                              <NavItem>
                                <div className=\\"nav-item\\">
                                  <NavLink className=\\"nav-link nav-right-side\\" isActive={[Function: isActive]} activeClassName=\\"nav-link active nav-right-side\\" to={{...}}>
                                    <Link aria-current=\\"page\\" className=\\"nav-link nav-right-side nav-link active nav-right-side\\" style={{...}} to={{...}}>
                                      <LinkAnchor aria-current=\\"page\\" className=\\"nav-link nav-right-side nav-link active nav-right-side\\" style={{...}} href=\\"/\\" navigate={[Function: navigate]}>
                                        <a aria-current=\\"page\\" className=\\"nav-link nav-right-side nav-link active nav-right-side\\" style={{...}} href=\\"/\\" onClick={[Function: onClick]}>
                                          Home
                                        </a>
                                      </LinkAnchor>
                                    </Link>
                                  </NavLink>
                                </div>
                              </NavItem>
                              <NavItem>
                                <div className=\\"nav-item\\">
                                  <NavLink className=\\"nav-link nav-right-side\\" activeClassName=\\"nav-link active nav-right-side\\" to={{...}}>
                                    <Link aria-current={{...}} className=\\"nav-link nav-right-side\\" style={[undefined]} to={{...}}>
                                      <LinkAnchor aria-current={{...}} className=\\"nav-link nav-right-side\\" style={[undefined]} href=\\"/media\\" navigate={[Function: navigate]}>
                                        <a aria-current={{...}} className=\\"nav-link nav-right-side\\" style={[undefined]} href=\\"/media\\" onClick={[Function: onClick]}>
                                          Media
                                        </a>
                                      </LinkAnchor>
                                    </Link>
                                  </NavLink>
                                </div>
                              </NavItem>
                              <NavItem>
                                <div className=\\"nav-item\\">
                                  <NavLink className=\\"nav-link nav-right-side\\" isActive={[Function: isActive]} activeClassName=\\"nav-link active nav-right-side\\" to={{...}}>
                                    <Link aria-current={{...}} className=\\"nav-link nav-right-side\\" style={[undefined]} to={{...}}>
                                      <LinkAnchor aria-current={{...}} className=\\"nav-link nav-right-side\\" style={[undefined]} href=\\"/login\\" navigate={[Function: navigate]}>
                                        <a aria-current={{...}} className=\\"nav-link nav-right-side\\" style={[undefined]} href=\\"/login\\" onClick={[Function: onClick]}>
                                          Profile
                                        </a>
                                      </LinkAnchor>
                                    </Link>
                                  </NavLink>
                                </div>
                              </NavItem>
                              <NavItem>
                                <div className=\\"nav-item\\">
                                  <NavLink className=\\"nav-link nav-right-end\\" activeClassName=\\"nav-link active\\" to={{...}}>
                                    <Link aria-current={{...}} className=\\"nav-link nav-right-end\\" style={[undefined]} to={{...}}>
                                      <LinkAnchor aria-current={{...}} className=\\"nav-link nav-right-end\\" style={[undefined]} href=\\"/login\\" navigate={[Function: navigate]}>
                                        <a aria-current={{...}} className=\\"nav-link nav-right-end\\" style={[undefined]} href=\\"/login\\" onClick={[Function: onClick]}>
                                          Login/Signup
                                        </a>
                                      </LinkAnchor>
                                    </Link>
                                  </NavLink>
                                </div>
                              </NavItem>
                            </div>
                          </ForwardRef>
                        </Nav>
                      </div>
                    </Transition>
                  </ForwardRef>
                </ForwardRef>
              </NavbarCollapse>
            </nav>
          </Navbar>
        </NavBar>
        <div className=\\"mainbody\\">
          <AlertBar history={{...}} location={{...}} match={{...}} updateAlertBar={[Function: updateAlertBar]} emptyBasket={[Function: emptyBasket]} addToBasket={[Function: addToBasket]} removeFromBasket={[Function: removeFromBasket]} getBasketItems={[Function: getBasketItems]} getTotalBasketPrice={[Function: getTotalBasketPrice]} checkBasketAvailabilities={[Function: checkBasketAvailabilities]} stripePromise={{...}} alert={{...}}>
            <Alert show={true} variant={[undefined]} onClose={[Function (anonymous)]} dismissible={true} transition={{...}} closeLabel=\\"Close alert\\">
              <Fade unmountOnExit={true} in={true} timeout={300} mountOnEnter={false} appear={false}>
                <ForwardRef addEndListener={[Function: transitionEndListener]} unmountOnExit={true} in={true} timeout={300} mountOnEnter={false} appear={false} onEnter={[Function (anonymous)]} childRef={{...}}>
                  <Transition unmountOnExit={true} in={true} timeout={300} mountOnEnter={false} appear={false} onEnter={[Function (anonymous)]} onEntered={[Function (anonymous)]} onEntering={[Function (anonymous)]} onExit={[Function (anonymous)]} onExited={[Function (anonymous)]} onExiting={[Function (anonymous)]} addEndListener={[Function (anonymous)]} nodeRef={{...}} enter={true} exit={true}>
                    <div role=\\"alert\\" className=\\"fade alert alert-dismissible show\\">
                      <CloseButton onClick={[Function (anonymous)]} aria-label=\\"Close alert\\" variant={[undefined]}>
                        <button type=\\"button\\" className=\\"btn-close\\" onClick={[Function (anonymous)]} aria-label=\\"Close alert\\" />
                      </CloseButton>
                    </div>
                  </Transition>
                </ForwardRef>
              </Fade>
            </Alert>
          </AlertBar>
          Attempting to refresh Stripe onboarding link...
        </div>
      </BasePage>
    </RefreshOnboardingStripeView>
  </Router>
</MemoryRouter>"
`;

exports[`RefreshOnboardingStripeView Should render correctly with shallow 1`] = `
"<BasePage history={{...}} location={{...}} match={{...}} updateAlertBar={[Function: updateAlertBar]} emptyBasket={[Function: emptyBasket]} addToBasket={[Function: addToBasket]} removeFromBasket={[Function: removeFromBasket]} getBasketItems={[Function: getBasketItems]} getTotalBasketPrice={[Function: getTotalBasketPrice]} checkBasketAvailabilities={[Function: checkBasketAvailabilities]} stripePromise={{...}} alert={{...}}>
  Attempting to refresh Stripe onboarding link...
</BasePage>"
`;
